.travis.yml

```diff
@@ -5,21 +5,30 @@ git:
 addons:
   apt:
     sources:
-    - sourceline: deb http://apt.llvm.org/precise/ llvm-toolchain-precise-3.8 main
-      key_url: http://apt.llvm.org/llvm-snapshot.gpg.key
-    - ubuntu-toolchain-r-test
-    - george-edison55-precise-backports
+      - sourceline: deb http://apt.llvm.org/precise/ llvm-toolchain-precise-3.8 main
+        key_url: http://apt.llvm.org/llvm-snapshot.gpg.key
+      - ubuntu-toolchain-r-test
+      - george-edison55-precise-backports
     packages:
-    - g++-6
-    - clang-3.8
-    - llvm-dev
+      - g++-6
+      - clang-3.8
+      - llvm-dev
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install gcc; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew upgrade python; fi
```

Comment:
this should only be done if the update succeeds. Let's use an `&&` to enforce that contingency